documentation/src/main/asciidoc/topics/query.adoc

```diff
@@ -18,480 +18,570 @@ Apart from indexed queries, {brandname} can run queries over non-indexed or
+=== Index Manager
 
-==== Configuration
+{brandname} uses internally a component called "Index Manager" to control how new data is applied to the index and
+when the data is visible to searches.
 
-===== General format
+The default Index Manager `directory-based` writes to the index as soon as the data is written to the cache. The downside is it can slow down considerably cache writes specially under heavy writing scenarios, since it needs to do constant costly operations called "flushes" on the index.
 
-To enable indexing via XML, you need to add the `<indexing>` element to your cache configuration, and optionally pass additional properties.
+The `near-real-time` index manager is similar to the default index manager but takes advantage of the Near-Real-Time features of Lucene. It has better write performance because it flushes the index to the underlying store less often.
+The drawback is that unflushed index changes can be lost in case of a non-clean shutdown. Can be used in conjunction with `local-heap` or `filesystem`.
 
-NOTE: The presence of an `<indexing>` element which omits the `enabled` attribute will auto-enable indexing for your
-convenience, even though the default value of the `enabled` attribute is defined as `"false"` in the XSD schema.
-This behaviour exists only on schemas starting with version 11.
+Example with `local-heap`:
 
 [source,xml,options="nowrap",subs=attributes+]
 ----
-include::config_examples/indexing.xml[]
+include::config_examples/indexing_near_real_time.xml[]
 ----
 
-Programmatic:
+Example with `filesystem`:
 
-[source,java]
+[source,xml,options="nowrap",subs=attributes+]
 ----
-import org.infinispan.configuration.cache.*;
-
-ConfigurationBuilder cacheCfg = ...
-cacheCfg.indexing().enable()
-      .addProperty("property name", "propery value")
+include::config_examples/indexing_near_real_time_fs.xml[]
 ----
 
-NOTE: When using programmatic configuration, indexing must be enabled explicitly by invoking the `enable()` method, as
-no auto-enabling mechanism exist for programmatic configuration as in the case of XML configuration.
-
-===== Index names
-
-Each property inside the `index` element is prefixed with the index name, for the index named `org.infinispan.sample.Car` the `directory_provider` is `local-heap`:
+[[query_massindexer]]
+=== Re-indexing
 
-[source,xml,options="nowrap",subs=attributes+]
-----
-include::config_examples/indexing_property.xml[]
-----
+Occasionally you might need to rebuild the indexs by reconstructing it from the data stored in the Cache. You need to rebuild the index if you change the definition of what is indexed on your types, or if you change for example some _Analyzer_ parameter, as Analyzers affect how the index is written. Also, you might need to rebuild the index if you had it destroyed by some system administration mistake. To rebuild the index just get a reference to the Indexer and start it; beware it might take some time as it needs to reprocess all data in the grid!
```

Comment:
s/indexs/indexes 

cache should be lowercase. I'd also try to cut down on the word count here. "Rebuilding an index reconstructs it from data stored in the cache. You need to rebuild indexes if you change things like definitions of indexed types or _Analyzer_ parameters. Likewise you might need to rebuild indexes if they are deleted for some reason."